src/games/strategy/triplea/delegate/MoveValidator.java

```diff
@@ -1430,8 +1430,10 @@ public static String validateCanal(final Route route, final Collection<Unit> uni
         continue;
       }
       failureMessage = canPassThroughCanal(canalAttachment, units, player, data);
-      if (!failureMessage.isPresent()) {
```

Comment:
Yes, that was a before of the method. You can easily add in the appropriate logic:
```
   return CanalAttachment.get(territory)
                         .stream()
                         .filter( canalAttachment ->  {
    return Properties.getControlAllCanalsBetweenTerritoriesToPass(data) && isCanalOnRoute(canalAttachment, route, data) })
                         .filter( canalAttachment ->  { return findErrorsWhenPassingThroughCanal(canalAttachment, units, player, data).isPresent() || ....)
                         .findFirst();
```